src/modules/transactions.js

```diff
@@ -0,0 +1,208 @@
+        id: tx.hash,
+        hash: tx.hash,
+        network: 'mainnet',
+        status: 'complete',
+        date: new Date(parseInt(tx.block_timestamp + '000', 10)),
+      };
+  
+      realm.write(() => {
+        realm.create('Transaction', newTx, 'modified');
+      });
+    };
+  
+    let data;
+    let page = 0;
+  
+    do {
+      data = await fetchTransactions({address: account, page});
+      data.transfers.forEach(handleTransaction);
+      page++;
+    } while (data.hasNextPage);
+  }
+
+  /**
+   * Load transactions for the current accounts
+   * 
+   * @returns transactions
+   */
+  async loadTransactions() {
+    const realm = getRealm();
+    const networkId = SubstrateNetwork.getInstance().networkId;
+
+    if (networkId === 'mainnet') {
+      const accounts = Accounts.getInstance().getAccounts();
+
+      for (const account of accounts) {
+        try {
+          await dispatch(
+            transactionsOperations.loadExternalTransactions(account.id),
+          );
+        } catch (err) {
+          console.error(err);
+        }
+      }
+    }
+
+    let items = realm.objects('Transaction').toJSON();
+
+    if (networkId === 'mainnet') {
+      items = items.filter(item => !(item.status === 'complete' && !item.hash));
+    }
+
+    return items;
+  }
+
+  /**
+   * Update transaction
+   * 
+   * @param {*} transaction 
+   */
+  updateTransaction(transaction){
+    const realm = getRealm();
+
+    realm.write(() => {
+      realm.create('Transaction', transaction, 'modified');
+    });
+  }
+
+  /**
+   * Get fee amount for the transaction
+   * @param {*} param0 
+   * @returns {int} fee amount
+   */
+  getFeeAmount({recipientAddress, accountAddress, amount}) {
+    return ApiRpc.getFeeAmount({
+      recipientAddress: recipientAddress,
+      accountAddress,
+      amount: amount,
+    });
+  }
+  
+  /**
+   * Send transaction
+   * @param {*} param0 
+   */
+  sendTransaction({recipientAddress, accountAddress, amount, fee, prevTransaction}) {
+    const parsedAmount = parseFloat(amount) * DOCK_TOKEN_UNIT;
+
+    const internalId = uuid();
+    const transaction = {
+      id: internalId,
+      date: new Date().toISOString(),
+      fromAddress: accountAddress,
+      recipientAddress: recipientAddress,
+      amount: `${parsedAmount}`,
+      feeAmount: `${fee}`,
+      status: TransactionStatus.InProgress,
+    };
+
+    const realm = getRealm();
+
+    realm.write(() => {
+      realm.create('Transaction', transaction, 'modified');
+    });
+
+    this.transactions.push(transaction);
+
+    return ApiRpc.sendTokens({
+      recipientAddress,
+      accountAddress,
+      amount: parsedAmount,
+    })
+      .then(res => {
+        const updatedTransation = {
+          ...transaction,
+          status: TransactionStatus.Complete,
+        };
+        this.updateTransaction(updatedTransation);
+
+        realm.write(() => {
+          realm.create('Transaction', updatedTransation, 'modified');
```

Comment:
is this a duplicate command to `this.updateTransaction(updatedTransaction)`?

Reply:
This implementation changed, seems fixed already.